byron/ledger/executable-spec/test/Ledger/Core/Generators/Properties.hs

```diff
@@ -54,6 +54,6 @@ relevantKValuesAreGenerated = withTests 500 $ property $ do
     --
     -- And if we round this value down we get 100 epochs.
 
-    cover 10
+    cover 6
```

Comment:
I'd argue it'd be useful to leave a comment next to changes like this one.

Reply:
(this change was unrelated to this PR, it was a setting that was cutting it too fine and happened to blow up now)

Reply:
My point is that we're lacking understanding why "cover 10" doesn't work and why lovering it to 6 works. It appears to be a black box, yet it shouldn't: try out a number and if it works, go with it. Again, this is not necessarily limited to your PR, but it could apply to coverage testing in general.

Reply:
it's true, these numbers are arbitrary and not exact, but I think they do give an indication of the shape and quality of traces and they tell us that our generation is matching with our testing to some extent at least

Reply:
>  try out a number and if it works, go with it. Again, this is not necessarily limited to your PR, but it could apply to coverage testing in general.

The way I did coverage test was to:

- write a coverage test
- check the figures we got out
  - if they made sense then lower the threshold by 10% to make sure we don't have brittle tests.
  - if they didn't make sense (for instance only 1% of self delegations), check the generators, and run the tests again.

Is this optimal? No. Are we aware of this? Yes. This is what we have at the moment, and what we could achieve in the time and resources we had. Rest assured that some of us are putting a lot of effort into this, and sometimes putting our personal time as well. Improvement proposals are always welcome.